org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/format/LSPFormatFilesHandler.java

```diff
@@ -81,15 +82,21 @@ protected void formatFile(final @NonNull IFile file, final IProgressMonitor moni
 				return;
 
 			monitor.setTaskName(NLS.bind(Messages.LSPFormatFilesHandler_FormattingFile, file.getFullPath()));
-			final VersionedFormatRequest formatRequest = formatter.versionedRequestFormatting(doc,
-					new TextSelection(0, 0));
+			final var formatting = formatter.requestFormatting(doc,
+					new TextSelection(0, 0), batchToken).get();
 
-			List<? extends TextEdit> formatEdits = formatRequest.edits().get();
-			if (formatEdits.isEmpty())
+			if (!formatting.isPresent())
 				return;
 
 			docProvider.aboutToChange(doc);
-			UI.getDisplay().syncExec(() -> formatter.applyEdits(doc, formatEdits, formatRequest.version()));
+			UI.getDisplay().syncExec(() -> {
+				try {
+					formatting.get().apply();
+				} catch (ConcurrentModificationException | BadLocationException e) {
+					LanguageServerPlugin.logError(e);
+
+				}
+			});
 			docProvider.changed(doc);
 			docProvider.saveDocument(monitor, file, doc, true);
```

Comment:
Since we have optional instead of return and then get() on the optional, I think it is better to use ifPresent:
```
			formatting.ifPresent(edits -> {
				docProvider.aboutToChange(doc);
				UI.getDisplay().syncExec(() -> {
					try {
						edits.apply();
					} catch (ConcurrentModificationException | BadLocationException e) {
						LanguageServerPlugin.logError(e);
					}
				});
				docProvider.changed(doc);
				try {
					docProvider.saveDocument(monitor, file, doc, true);
				} catch (CoreException e) {
					LanguageServerPlugin.logError(e);
				}
			});

```